### PR TITLE
SHYAML

### PR DESCRIPTION
**Pull request by scopatz**
annoying noarch python recipe

### CODE REVIEW

**Comment by scopatz** [comment]
@conda-forge/staged-recipes - this is ready for review!

File: recipes/shyaml/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 80650ebfe6fa2e16083451d515207472d60990c1c15fc0fd607c27077790ac23
+  patches:
+    - cli-args.patch
+
+build:
+  noarch: python
+  number: 0
+  #script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv"
```
Comment (jakirkham) [inline recipes/shyaml/meta.yaml]: Guess this line can be dropped (or moved below?).

File: recipes/shyaml/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 80650ebfe6fa2e16083451d515207472d60990c1c15fc0fd607c27077790ac23
+  patches:
+    - cli-args.patch
```
Comment (jakirkham) [inline recipes/shyaml/meta.yaml]: Could we add a comment above this about what it is needed for? Also has this been raised with upstream?

Comment (scopatz) [reply recipes/shyaml/meta.yaml]: I think this is an issue only for entry points, which the original package doesn't use but we have to.

File: recipes/shyaml/meta.yaml (hunk 1 above)
Comment (scopatz) [reply recipes/shyaml/meta.yaml]: Yep, thanks!  Changes made

File: recipes/shyaml/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  sha256: 80650ebfe6fa2e16083451d515207472d60990c1c15fc0fd607c27077790ac23
+  patches:
+    # this is needed to make conda entry point work.
+    - cli-args.patch
+
+build:
+  noarch: python
+  number: 0
+  script: |
+    $PYTHON setup.py install
```
Comment (jakirkham) [inline recipes/shyaml/meta.yaml]: Since we are using `setuptools`, do we need the other arguments here? Was there a reason `pip` didn't work for this?

**Comment by jakirkham** [comment]
Thanks @scopatz. Looks pretty good. Couple minor comments.

File: recipes/shyaml/meta.yaml (hunk 3 above)
Comment (scopatz) [reply recipes/shyaml/meta.yaml]: Yeah pip fails to install for some reason, might be because it is using `d2to1` and other nonstandard hacks

**Comment by scopatz** [comment]
I believe that this is ready @conda-forge/staged-recipes

**Pull request merged**